### PR TITLE
Update error.xml

### PR DESCRIPTION
**Pull request by AurelioDeRosa**
Fix issue #413.

### CODE REVIEW

File: entries/error.xml [hunk 1]
```diff
@@ -52,6 +52,15 @@ $( "img" )
   })
   .attr( "src", "missing.png" );
 ]]></code>
+    <p>In the example, missing.png is an hypotetical missing image that is used to trigger the <code>error</code> event, not an image to replace those giving an error. To do that, you have to modify the body of the anonymous function passed to <code>.error()</code> as shown below:</p>
```
Comment (dmethvin) [inline entries/error.xml]: hypothetical

File: entries/error.xml [hunk 2]
```diff
@@ -52,6 +52,15 @@ $( "img" )
...
+    <code><![CDATA[
+$( "img" )
+  .error(function() {
+    $( this ).attr( "src", "replacement.png" );
+  })
+  .attr( "src", "missing.png" );
+]]></code>
+  <p><b>Note:</b> Be sure that the replacement image exists otherwise the <code>error</code> event will be triggered indefinitely.</p>
```
Comment (dmethvin) [inline entries/error.xml]: Good note!

File: entries/error.xml (hunk 1 above)
Comment (AurelioDeRosa) [inline entries/error.xml]: Typo corrected. Sorry about that.

File: entries/error.xml (hunk 2 above)
Comment (kswedberg) [inline entries/error.xml]: Please add a semicolon after "exists"

Comment (AurelioDeRosa) [inline entries/error.xml]: Done :)

**Comment by arthurvr** [comment]
This PR looks good to me. Shall I merge?

File: entries/error.xml [hunk 3]
```diff
@@ -52,6 +52,15 @@ $( "img" )
...
 ]]></code>
+    <p>In the example, missing.png is an hypothetical missing image that is used to trigger the <code>error</code> event, not an image to replace those giving an error. To do that, you have to modify the body of the anonymous function passed to <code>.error()</code> as shown below:</p>
+    <code><![CDATA[
+$( "img" )
+  .error(function() {
+    $( this ).attr( "src", "replacement.png" );
+  })
+  .attr( "src", "missing.png" );
+]]></code>
+  <p><b>Note:</b> Be sure that the replacement image exists; otherwise the <code>error</code> event will be triggered indefinitely.</p>
```
Comment (arthurvr) [inline entries/error.xml]: the indentation for this line seems wrong.

Comment (arthurvr) [inline entries/error.xml]: Also, I don't think `<p>` tags are allowed in `<example>`s. This just results into this: http://gyazo.com/f7795ca0fd998b6701b68bbb322fed0d (notice your extra info inside the `<p>` isn't rendered).

@AurelioDeRosa Do you have a working local docs setup? Normally you should be able to repro that there. If you don't, ping me on IRC and I'll  be happy to help you out.

Comment (arthurvr) [inline entries/error.xml]: ping @AurelioDeRosa

Comment (AurelioDeRosa) [inline entries/error.xml]: @arthurvr I'm going to take care of this issue right now. Sorry for the delay.

**Comment by AurelioDeRosa** [comment]
As @arthurvr noted, the current version of the XSL file doesn't allow to have a paragraph in an `example`. So, the only solution I see is to allow it. Who can fix this issue? Any alternatives? // cc @dmethvin @kswedberg @scottgonzalez

**Comment by dmethvin** [comment]
I think that notes were intended for small notes and not big clips with paragraphs and examples. If we allow big clips and block elements like code, it may affect layout in a drastic way so that's something to take into consideration.

**Comment by scottgonzalez** [comment]
Well you for sure can't just throw arbitrary markup into `<example>`. We can potentially add support for `<longdesc>` inside `<example>`, but the currently used markup will never be allowed.

**Comment by AurelioDeRosa** [comment]
What's your suggestion to rearrange the code to have a description?

**Comment by scottgonzalez** [comment]
That's hard to say because I can't tell what you're trying to do. Are you trying to create an inline runnable example that is accompanied by a code block explaining how to do something? This is very different from the way any other page is designed.

**Comment by arthurvr** [comment]
In this case I'd opt to move some stuff in the `<desc>` and other things can fit in a JavaScript comment.

> We can potentially add support for <longdesc> inside <example>, but the currently used markup will never be allowed.

This is kinda off-topic for this PR, and should be discussed in `grunt-jquery-content` isn't it?

**Comment by scottgonzalez** [comment]
> In this case I'd opt to move some stuff in the <desc> and other things can fit in a JavaScript comment.

The `<desc>` is really a title. So I don't think that's an acceptable solution.

> >  We can potentially add support for inside , but the currently used markup will never be allowed.

> This is kinda off-topic for this PR, and should be discussed in grunt-jquery-content isn't it?

We certainly shouldn't open an issue for grunt-jquery-content until we know what we're actually discussing. I'd like to hear from @AurelioDeRosa since he sent the PR which contains two `<code>` blocks inside `<example>`, along with other HTML content.

**Comment by AurelioDeRosa** [comment]
@scottgonzalez The PR addresses #413. The idea is to have a description of what's going on in the example for the benefit of some users.

**Comment by scottgonzalez** [comment]
Sounds like what you really want is a new, useful example.

**Comment by AurelioDeRosa** [comment]
I'd not say that a new example will solve the problem. The problem was the user didn't get what the example really did. Perhaps @arthurvr's suggestion of having a few comments in the code can fix this.

**Comment by AurelioDeRosa** [comment]
@arthurvr @scottgonzalez Should I try to submit a PR which has JavaScript comment so that we can review it?

**Comment by scottgonzalez** [comment]
sure

**Comment by arthurvr** [comment]
I'd love to move forward on this PR. ping @AurelioDeRosa

**Comment by AurelioDeRosa** [comment]
PR updated @arthurvr.

File: entries/error.xml [hunk 4]
```diff
@@ -20,7 +20,7 @@
     </argument>
   </signature>
   <longdesc>
-    <p>This method is a shortcut for <code>.on( "error", handler )</code>.</p>
+    <p><b>As of jQuery 1.8</b>, the <code>.error()</code> method is deprecated. Use <code>.on( "error", handler )</code> to attach event handlers to the <code>error</code> event instead.</p>
```
Comment (arthurvr) [inline entries/error.xml]: This is a totally unrelated change. Mind extracting this to its own commit?

Comment (arthurvr) [inline entries/error.xml]: ping @AurelioDeRosa ;)

Comment (arthurvr) [inline entries/error.xml]: hey-hey @AurelioDeRosa still interested in picking this up?

Comment (arthurvr) [inline entries/error.xml]: ping @AurelioDeRosa

**Comment by AurelioDeRosa** [comment]
PR updated @arthurvr. Sorry for the late change.

**Comment by arthurvr** [comment]
@AurelioDeRosa I liked the change, it was just unrelated to the thing you're trying to fix, so it should have been in a separated commit. I didn't mean you should remove the change :wink:

**Comment by AurelioDeRosa** [comment]
@arthurvr I was thinking about opening a different PR for it, that's why I removed it.

**Comment by arthurvr** [comment]
Makes sense. Thanks.

File: entries/error.xml [hunk 5]
```diff
@@ -20,7 +20,6 @@
     </argument>
   </signature>
   <longdesc>
-    <p>This method is a shortcut for <code>.on( "error", handler )</code>.</p>
```
Comment (arthurvr) [inline entries/error.xml]: This paragraph shouldn't be removed (or at least not in this commit).

Comment (AurelioDeRosa) [inline entries/error.xml]: Oh... 🐑, you're right!

**Comment by AurelioDeRosa** [comment]
Can someone check the updated PR so that we can merge it?

File: entries/error.xml [hunk 6]
```diff
@@ -34,18 +34,19 @@ $( "#book" )
...
     </div>
-    <p>Note: A jQuery error event handler should not be attached to the window object. The browser fires the window's error event when a script error occurs. However, the window error event receives different arguments and has different return value requirements than conventional event handlers. Use <code>window.onerror</code> instead.</p>
+    <p>Note: A jQuery <code>error</code> event handler should not be attached to the <code>window</code> object. The browser fires the <code>window</code>'s <code>error</code> event when a script error occurs. However, the window <code>error</code> event receives different arguments and has different return value requirements than conventional event handlers. Use <code>window.onerror</code> instead.</p>
   </longdesc>
   <example>
-    <desc>To hide the "broken image" icons for IE users, you can try:</desc>
+    <desc>To hide the "broken image" icons for users of older versions of IE, you can write:</desc>
```
Comment (arthurvr) [inline entries/error.xml]: > of older versions of IE

Which versions does that affect?

Comment (AurelioDeRosa) [inline entries/error.xml]: I remember is IE <= 10 but I can't find any reference. Someone with a hint on this?

Comment (arthurvr) [inline entries/error.xml]: No idea, honestly. I do think though that just saying "older versions of IE" means nothing so we should figure this out. Maybe @dmethvin or @timmywil have links to relevant discussions about this?

Comment (arthurvr) [inline entries/error.xml]: ping @AurelioDeRosa @dmethvin @timmywil is there anyone who knows this? If not I'll quickly test this out.

Comment (dmethvin) [inline entries/error.xml]: Yeah, I mentioned elsewhere that I'm not sure when the "broken image" thing went away. Probably before IE9?

Comment (AurelioDeRosa) [inline entries/error.xml]: Should we specify IE9 for now and update in case of mistake?

**Comment by AurelioDeRosa** [comment]
This PR is standing from a few months. Is anyone happy to specify IE9 and merge it?

**Comment by dmethvin** [comment]
Looks like this has been settable from at least IE and Firefox for years:

http://stackoverflow.com/questions/1694937/firefox-show-broken-images
http://ccm.net/faq/13546-internet-explorer-show-image-download-placeholders

However, I couldn't find out when the default was changed. How about if we just delete the example? There is still a good one there.

**Comment by AurelioDeRosa** [comment]
@dmethvin Thank you for the feedback. I've updated the PR based on your comment.

**Comment by arthurvr** [comment]
lgtm :+1:
